Document what fields in the Channel structure work with what channel types (#5708)
hammerandchisel/discord-api-docs: docs/resources/Channel.md

| type                                | integer                                                                     | the [type of channel](#DOCS_RESOURCES_CHANNEL/channel-object-channel-types)                                                                                                                   | ALL                                                                                                                                                                |
| guild_id?                           | snowflake                                                                   | the id of the guild (may be missing for some channel objects received over gateway guild dispatches)                                                                                          | GUILD_TEXT, GUILD_VOICE, GUILD_CATEGORY, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, GUILD_STAGE_VOICE, GUILD_DIRECTORY, FORUM_CHANNEL |

> | position?                           | integer                                                                     | sorting position of the channel                                                                                                                                                               | GUILD_TEXT, GUILD_VOICE, GUILD_CATEGORY, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, GUILD_STAGE_VOICE, GUILD_DIRECTORY, FORUM_CHANNEL |

Reviewer: ```suggestion
| position?                           | integer                                                                     | sorting position of the channel                                                                                                                                                               | GUILD_TEXT, GUILD_VOICE, GUILD_CATEGORY, GUILD_ANNOUNCEMENT, GUILD_STAGE_VOICE, GUILD_DIRECTORY, FORUM_CHANNEL |
```

| permission_overwrites?              | array of [overwrite](#DOCS_RESOURCES_CHANNEL/overwrite-object) objects      | explicit permission overwrites for members and roles                                                                                                                                          | GUILD_TEXT, GUILD_VOICE, GUILD_CATEGORY, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, GUILD_STAGE_VOICE, GUILD_DIRECTORY, FORUM_CHANNEL |

> | name?                               | ?string                                                                     | the name of the channel (1-100 characters)                                                                                                                                                    | GUILD_TEXT, GUILD_VOICE, GUILD_CATEGORY, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, GUILD_STAGE_VOICE, GUILD_DIRECTORY, FORUM_CHANNEL |

Reviewer: ```suggestion
| name?                               | ?string                                                                     | the name of the channel (1-100 characters)                                                                                                                                                    | GROUP_DM, GUILD_TEXT, GUILD_VOICE, GUILD_CATEGORY, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, GUILD_STAGE_VOICE, GUILD_DIRECTORY, FORUM_CHANNEL |
```

| topic?                              | ?string                                                                     | the channel topic (0-4096 characters for `GUILD_FORUM` channels, 0-1024 characters for all others)                                                                                            | GUILD_TEXT, GUILD_ANNOUNCEMENT, FORUM_CHANNEL                                                                                                                      |
| nsfw?                               | boolean                                                                     | whether the channel is nsfw                                                                                                                                                                   | GUILD_TEXT, GUILD_VOICE, GUILD_ANNOUNCEMENT, FORUM_CHANNEL                                                                                                         |

> | last_message_id?                    | ?snowflake                                                                  | the id of the last message sent in this channel (or thread for `GUILD_FORUM` channels) (may not point to an existing or valid message or thread)                                              | DM, GROUP_DM, GUILD_TEXT, GUILD_VOICE, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, GUILD_STAGE_VOICE, GUILD_DIRECTORY, FORUM_CHANNEL                 |

Reviewer 2: Isn't this every channel?

Author: Categories don't implement this value.

> | last_message_id?                    | ?snowflake                                                                  | the id of the last message sent in this channel (or thread for `GUILD_FORUM` channels) (may not point to an existing or valid message or thread)                                              | GUILD_TEXT, GUILD_VOICE, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, GUILD_STAGE_VOICE, GUILD_DIRECTORY, FORUM_CHANNEL                 |

Reviewer: ```suggestion
| last_message_id?                    | ?snowflake                                                                  | the id of the last message sent in this channel (or thread for `GUILD_FORUM` channels) (may not point to an existing or valid message or thread)                                              | DM, GROUP_DM, GUILD_TEXT, GUILD_VOICE, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, GUILD_STAGE_VOICE, GUILD_DIRECTORY, FORUM_CHANNEL                 |
```

| owner_id?                           | snowflake                                                                   | id of the creator of the group DM or thread                                                                                                                                                   | GROUP_DM, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD                                                                                                       |
| application_id?                     | snowflake                                                                   | application id of the group DM creator if it is bot-created                                                                                                                                   | GROUP_DM                                                                                                                                                           |

> | parent_id?                          | ?snowflake                                                                  | for guild channels: id of the parent category for a channel (each parent category can contain up to 50 channels), for threads: id of the text channel this thread was created                 | DM, GUILD_VOICE, GUILD_CATEGORY, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, GUILD_STAGE_VOICE, FORUM_CHANNEL                          |

Reviewer: ```suggestion
| parent_id?                          | ?snowflake                                                                  | for guild channels: id of the parent category for a channel (each parent category can contain up to 50 channels), for threads: id of the text channel this thread was created                 | GUILD_TEXT, GUILD_VOICE, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, GUILD_STAGE_VOICE, FORUM_CHANNEL                          |
```

> | last_pin_timestamp?                 | ?ISO8601 timestamp                                                          | when the last pinned message was pinned. This may be `null` in events such as `GUILD_CREATE` when a message is not pinned.                                                                    | DM, GUILD_ANNOUNCEMENT, FORUM_CHANNEL                                                                                                                              |

Reviewer 3: ```suggestion
| last_pin_timestamp?                 | ?ISO8601 timestamp                                                          | when the last pinned message was pinned. This may be `null` in events such as `GUILD_CREATE` when a message is not pinned.                                                                    | DM, GUILD_ANNOUNCEMENT, FORUM_CHANNEL, GUILD_TEXT, PUBLIC_THREAD, PRIVATE_THREAD                                                                                                                            |
```
those have pins too

> | rtc_region?                         | ?string                                                                     | [voice region](#DOCS_RESOURCES_VOICE/voice-region-object) id for the voice channel, automatic when set to null                                                                                | GUILD_VOICE                                                                                                                                                        |

Reviewer: ```suggestion
| rtc_region?                         | ?string                                                                     | [voice region](#DOCS_RESOURCES_VOICE/voice-region-object) id for the voice channel, automatic when set to null                                                                                | GUILD_VOICE, GUILD_STAGE_VOICE                                                                                                                                                        |
```

> | video_quality_mode?                 | integer                                                                     | the camera [video quality mode](#DOCS_RESOURCES_CHANNEL/channel-object-video-quality-modes) of the voice channel, 1 when not present                                                          | GUILD_VOICE                                                                                                                                                        |

Reviewer: ```suggestion
| video_quality_mode?                 | integer                                                                     | the camera [video quality mode](#DOCS_RESOURCES_CHANNEL/channel-object-video-quality-modes) of the voice channel, 1 when not present                                                          | GUILD_VOICE, GUILD_STAGE_VOICE                                                                                                                                                        |
```

| flags?                              | integer                                                                     | [channel flags](#DOCS_RESOURCES_CHANNEL/channel-object-channel-flags) combined as a [bitfield](https://en.wikipedia.org/wiki/Bit_field)                                                       | 0, 2, 4, 5, 10, 11, 12, 13, 14, 15 |
| total_message_sent?                 | integer                                                                     | number of messages ever sent in a thread, it's similar to `message_count` on message creation, but will not decrement the number when a message is deleted                                    | 10, 11, 12                         |
| available_tags?                     | array of [tag](#DOCS_RESOURCES_CHANNEL/forum-tag-object) objects            | the set of tags that can be used in a `GUILD_FORUM` channel                                                                                                                                   | 15                                 |

> | applied_tags?                       | array of snowflakes                                                         | the IDs of the set of tags that have been applied to a thread in a `GUILD_FORUM` channel                                                                                                      | 15                                 |

Reviewer 4: Applied tags are on threads, not on forum channels.

Author: oop

> | flags?                              | integer                                                                     | [channel flags](#DOCS_RESOURCES_CHANNEL/channel-object-channel-flags) combined as a [bitfield](https://en.wikipedia.org/wiki/Bit_field)                                                       | GUILD_TEXT, GUILD_VOICE, GUILD_CATEGORY, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, GUILD_STAGE_VOICE, GUILD_DIRECTORY, FORUM_CHANNEL |

Reviewer: ```suggestion
| flags                               | integer                                                                     | [channel flags](#DOCS_RESOURCES_CHANNEL/channel-object-channel-flags) combined as a [bitfield](https://en.wikipedia.org/wiki/Bit_field)                                                       | ALL                                                                                                                                                                |
```

Author: Pretty sure flags isn't optional

> | available_tags?                     | array of [tag](#DOCS_RESOURCES_CHANNEL/forum-tag-object) objects            | the set of tags that can be used in a `GUILD_FORUM` channel                                                                                                                                   | PUBLIC_THREAD                                                                                                                                                      |

Reviewer 3: ```suggestion
| available_tags?                     | array of [tag](#DOCS_RESOURCES_CHANNEL/forum-tag-object) objects            | the set of tags that can be used in a `GUILD_FORUM` channel                                                                                                                                   | GUILD_FORUM                                                                                                                                                      |
```

> | applied_tags?                       | array of snowflakes                                                         | the IDs of the set of tags that have been applied to a thread in a `GUILD_FORUM` channel                                                                                                      | FORUM_CHANNEL                                                                                                                                                      |

Reviewer 3: ```suggestion
| applied_tags?                       | array of snowflakes                                                         | the IDs of the set of tags that have been applied to a thread in a `GUILD_FORUM` channel                                                                                                      | PUBLIC_THREAD                                                                                                                                                      |
```

| default_reaction_emoji?             | ?[default reaction](#DOCS_RESOURCES_CHANNEL/default-reaction-object) object | the emoji to show in the add reaction button on a thread in a `GUILD_FORUM` channel                                                                                                           | FORUM_CHANNEL                                                                                                                                                      |

> | default_thread_rate_limit_per_user? | integer                                                                     | the initial `rate_limit_per_user` to set on newly created threads in a channel. this field is copied to the thread at creation time and does not live update.                                 | GUILD_TEXT, GUILD_ANNOUNCEMENT, 10, PUBLIC_THREAD, PRIVATE_THREAD, FORUM_CHANNEL                                                                                   |

Reviewer 3: ```suggestion
| default_thread_rate_limit_per_user? | integer                                                                     | the initial `rate_limit_per_user` to set on newly created threads in a channel. this field is copied to the thread at creation time and does not live update.                                 | GUILD_TEXT, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, FORUM_CHANNEL                                                                                   |
```

> | default_thread_rate_limit_per_user? | integer                                                                     | the initial `rate_limit_per_user` to set on newly created threads in a channel. this field is copied to the thread at creation time and does not live update.                                 | GUILD_TEXT, GUILD_ANNOUNCEMENT, ANNOUNCEMENT_THREAD, PUBLIC_THREAD, PRIVATE_THREAD, FORUM_CHANNEL                                                                  |

Reviewer: ```suggestion
| default_thread_rate_limit_per_user? | integer                                                                     | the initial `rate_limit_per_user` to set on newly created threads in a channel. this field is copied to the thread at creation time and does not live update.                                 | GUILD_TEXT, GUILD_ANNOUNCEMENT, FORUM_CHANNEL                                                                  |
```

Author: The thread rate limit parameter is copied into thread objects.

Reviewer: It's copied to `rate_limit_per_user`.
